### ISSUE TITLE
Implicit declaration of pthread_setconcurrency()

### ISSUE THREAD

**Issue by kamar535** [os-assignments/threads-synchronization-deadlock#2]
Compiling higher-grade on some Linux distributions:

```
cd higher-grade
make
```
, gives the following error:

```
src/n_barrier_test.c: In function ‘main’:
src/n_barrier_test.c:98:5: error: implicit declaration of function ‘pthread_setconcurrency’ [-Werror=implicit-function-declaration]
   98 |     pthread_setconcurrency(NTHREADS);
      |     ^~~~~~~~~~~~~~~~~~~~~~
```

**Comment by kamar535** [comment]
The pthread_setconcurrency() function does not seem to be part of all implementations of Pthreads.

I don't think this line is strictly needed. If you get this error, simply comment out this line.

**Comment by kamar535** [comment]
Fixed by 9b976ac9b52e08c204df117759d6e6124aa898c6

**Comment by kamar535** [comment]
According to [POSIX.1-2017](https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/), the `pthread_getconcurrency()` and `pthread_setconcurrency()` functions are marked [obsolescent](https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_getconcurrency.html).
